src/thor/trace_route_action.cc

```diff
@@ -151,34 +151,36 @@ void thor_worker_t::route_match(Api& request) {
 // of each edge. We will need to use the existing costing method to form the elapsed time
 // the path. We will start with just using edge costs and will add transition costs.
 std::vector<std::tuple<float, float, std::vector<thor::MatchResult>>>
-thor_worker_t::map_match(Api& request, uint32_t best_paths) {
+thor_worker_t::map_match(Api& request) {
   auto& options = *request.mutable_options();
   // Call Meili for map matching to get a collection of Location Edges
   matcher->set_interrupt(interrupt);
   // Create the vector of matched path results
   if (trace.size() == 0) {
     return {};
   }
-  m_offline_results = matcher->OfflineMatch(trace, best_paths);
+
+  // we don't allow multi path for trace route
+  // TODO::we have to get back for alternitives for trace route
+  int best_path =
+      request.options().action() == Options::trace_attributes ? request.options().best_paths() : 1;
+  m_offline_results = matcher->OfflineMatch(trace, best_path);
 
   // Process each score/match result
   for (const auto& result : m_offline_results) {
     const auto& match_results = result.results;
     const auto& edge_segments = result.segments;
     m_temp_enhanced_match_results.clear();
     m_temp_route_discontinuities.clear();
+    m_temp_disconnected_edges.clear();
 
     // Form the path edges based on the matched points and populate disconnected edges
-    m_temp_disconnected_edges.clear();
     m_temp_path_edges = MapMatcher::FormPath(matcher.get(), match_results, edge_segments,
                                              mode_costing, mode, m_temp_disconnected_edges, options);
 
-    // Throw exception if not trace attributes action and disconnected path.
-    // TODO - perhaps also throw exception if use_timestamps and disconnected path?
-    if (options.action() == Options::trace_route &&
-        (!m_temp_disconnected_edges.empty() || m_temp_path_edges.empty())) {
+    if (m_temp_path_edges.empty() && options.action() == Options::trace_route) {
```

Comment:
we should try not to delete comments from the code. as you know this stuff is complicated. we need to keep comments in so that when the next person gets to the code its easier for them to understand what is going on. here we are throwing if you are doing trace route and we found no route at all.